Add Drain Cleaner installation files to the operators repo

scholzj opened the pull request:
### Type of change

- Task

### Description

As described in the [SP-29](https://github.com/strimzi/proposals/blob/main/029-adopt-the-drain-cleaner-utility.md), the installation files for the Drain Cleaner project should be added to the operators installation folders. It also adds a README.md file linking it to the Drain Cleaner repository / README.md. In the future, this should probably link to the Strimzi docs.

This PR adds it only to the `packaging/install` folder => it will propagate to the `install/` folder in the next release.

tombentley commented on packaging/install/drain-cleaner/kubernetes/040-Secret.yaml:
```diff
@@ -0,0 +1,11 @@
+  namespace: strimzi-drain-cleaner
+type: kubernetes.io/tls
+data:
+  tls.crt: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUVsVENDQTMyZ0F3SUJBZ0lVZFVFeFR3UjFXQkFKc29YQzhaeWdkd1gySGV3d0RRWUpLb1pJaHZjTkFRRUwKQlFBd1hqRUxNQWtHQTFVRUJoTUNRMW94RHpBTkJnTlZCQWNUQmxCeVlXZDFaVEVlTUJ3R0ExVUVDaE1WVTNSeQphVzE2YVNCRWNtRnBiaUJEYkdWaGJtVnlNUjR3SEFZRFZRUURFeFZUZEhKcGJYcHBSSEpoYVc1RGJHVmhibVZ5ClEwRXdIaGNOTWpFd01USXpNakF6TkRBd1doY05NekF3TXpFeE1EUXpOREF3V2pCa01Rc3dDUVlEVlFRR0V3SkQKV2pF ...
+  tls.key: LS0tLS1CRUdJTiBQUklWQVRFIEtFWS0tLS0tCk1JSUV2UUlCQURBTkJna3Foa2lHOXcwQkFRRUZBQVNDQktjd2dnU2pBZ0VBQW9JQkFRQzJCSFc1TkhaYVlIM0cKMHZLMEV6dWFQTkR2cUhKT1dadmNUNFhWVW5KaFN6T1d4MTFuZnlzNUVYRHFiYmJWck9aVzNvMmgwUEVhM0YweQpzQ21tem1qMUJ3dksrUnhXM1diSUxNM1FFQUNPMkI1ZDMyMTB0a1VIZk9UT2ZESS9CS2QwK3JvOUs5NUVIMkU3Cm95VnhhK2NhaTgvSFJpRGI1ZVFFQXUxclgwOHdOeW1GWkpVeGtHN3VNNmRydjZiWEhSYXdDRUpBNVZicHdZZlkKbmJQ ...
```
Are these hard coded cert+key?

scholzj replied:
Yes, this is the prepared key for the pre-defined namespace. If you would not have it there, everyone would need to first generate the key and update all the resources.

tombentley commented on packaging/install/drain-cleaner/kubernetes/README.md:
```diff
@@ -0,0 +1,13 @@
+# Installing Strimzi Drain Cleaner on Kubernetes without CertManager
+
+Strimzi Drain Cleaner is using `ValidationWebhook` to receive information about Strimzi pods being evicted.
```
```suggestion
Strimzi Drain Cleaner uses a `ValidationWebhook` to receive information about Strimzi pods being evicted.
```

tombentley commented on packaging/install/drain-cleaner/kubernetes/README.md:
```diff
@@ -0,0 +1,13 @@
+
+Strimzi Drain Cleaner is using `ValidationWebhook` to receive information about Strimzi pods being evicted.
+Kubernetes requires that `ValidationWebhooks` are secured using TLS.
+So the web-hook service need to have HTTPS support.
```
```suggestion
So the web-hook service needs to have HTTPS support.
```

tombentley commented on packaging/install/drain-cleaner/kubernetes/webhook-certificates/build.sh:
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
```
Should we check for the existence of `cfssl` and exit with a message if it's missing?

tombentley commented on packaging/install/drain-cleaner/kubernetes/webhook-certificates/build.sh:
```diff
@@ -0,0 +1,16 @@
+cat ca.pem >> webhook-bundle.crt
+
+# Convert keys to PKCS8
+openssl pkcs8 -topk8 -nocrypt -in ca-key.pem -out ca.key
```
Do we have to depend on _both_ cfssl and openssl?

scholzj replied:
It is just an example script ... at the end everyone can generate the certificates anyway they want. They do not have to use these tools.

tombentley reviewed (approved):
A few comments, but basically LGTM.

scholzj commented:
@tombentley I fixed the README.md and added the checks for the installed programs. I will also open a PR in the Drain Cleaner repo.

I understand that the content of the `kubernetes` folder is not optimal, but I think that having the example including the scripts and certificates is the best possible way to show users how to install it (for those who are not on OCP or do not have Cert Manager installed). If you have some better ideas, we should discuss them.

The script as it is here is as it was released - so I think if we wanna do some further changes to it, we should do it in the Drain Cleaner repo for the next Drain Cleaner release.

WDYT?
